lib/galaxy/objectstore/irods.py

```diff
@@ -406,10 +395,10 @@ def _push_to_irods(self, rel_path, source_file=None, from_string=None):
             # Create sub-collection first
             self.session.collections.create(collection_path, recurse=True)
 
-            # Create data object
-            data_obj = self.session.data_objects.create(data_object_path, self.resource, **options)
-
             if from_string:
+                # Create data object
```

Comment:
This is needed only in this path through the code. 
Also, I do not see any callers, where 'from_string' is set. Can this if clause be removed?

Reply:
Yes, you can remove it.